### PR TITLE
Fix CORS header missing in 404 responses for unknown URLs (fixes #414)

### LINKED ISSUE #414: CORS is missing on unknown URL (404)
CORS are present when a 404 occurs from a resource missing record.

However when the URL is unknown, the ``reapply_cors`` method has no effect (since ``utils.current_service()`` returns ``None``)

Note: This was reported here https://github.com/mozilla-services/syncto/issues/5

### PR DESCRIPTION
**Pull request by leplatrem**
@ametaireau r?

### CODE REVIEW

File: cliquet/utils.py [hunk 1]
```diff
@@ -156,6 +156,15 @@ def reapply_cors(request, response):
     if service:
         request.info['cors_checked'] = False
         response = cors.ensure_origin(service, request, response)
+    else:
+        # No existing service is concerned, and Cornice is not implied.
+        origin = request.headers.get('Origin')
+        if origin:
+            allowed_origins = request.registry.settings['cliquet.cors_origins']
+            origins_match = decode_header(origin) == allowed_origins
```
Comment (Natim) [inline cliquet/utils.py]: I would add parenthesis to be more readable.

File: cliquet/utils.py [hunk 2]
```diff
@@ -156,6 +156,15 @@ def reapply_cors(request, response):
     if service:
         request.info['cors_checked'] = False
         response = cors.ensure_origin(service, request, response)
+    else:
+        # No existing service is concerned, and Cornice is not implied.
+        origin = request.headers.get('Origin')
+        if origin:
+            allowed_origins = request.registry.settings['cliquet.cors_origins']
```
Comment (Natim) [inline cliquet/utils.py]: I think cors_origins should be a list rather than a value.

File: cliquet/utils.py [hunk 3]
```diff
@@ -156,6 +156,15 @@ def reapply_cors(request, response):
...
+            origins_match = (decode_header(origin) in allowed_origins)
+            if allowed_origins == '*' or origins_match:
```
Comment (Natim) [inline cliquet/utils.py]: foobar = Service(name='foobar', path='/foobar', cors_origins=('*',))

So we should probably have ``origins_match = (decode_header(origin) in allowed_origins or '*' in allowed_origins)``

Comment (ametaireau) [inline cliquet/utils.py]: See https://github.com/mozilla-services/cornice/blob/master/cornice/cors.py#L102 to reproduce the same behaviour.

**Comment by ametaireau** [comment]
r+ wc

**Comment by Natim** [comment]
r+

**Pull request merged**